### PR TITLE
fix devShell for rust-analyzer

### PR DESCRIPTION
**Pull request by milahu**
fix the rust-analyzer extension in vscode

error was

> rust-analyzer failed to load workspace: Failed to find sysroot for Cargo.toml ... Is rust-src installed?

based on
https://discourse.nixos.org/t/rust-src-not-found-and-other-misadventures-of-developing-rust-on-nixos/11570/2

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -27,8 +27,10 @@
           nativeBuildInputs = with pkgs; [
             rustc
             cargo
+            rustPlatform.rustLibSrc
```
Comment (Mic92) [inline flake.nix:30]: What does adding it to buildInputs do?

Comment (milahu) [reply flake.nix:30]: nothing i guess ...

File: flake.nix [hunk 2]
```diff
@@ -27,8 +27,10 @@
...
             gitAndTools.pre-commit
           ];
+          RUST_SRC_PATH = "${pkgs.rustPlatform.rustLibSrc}";
```
Comment (Artturin) [inline flake.nix:33]:
```suggestion
          RUST_SRC_PATH = pkgs.rustPlatform.rustLibSrc;
```

File: flake.nix (hunk 1 above)
Comment (Artturin) [reply flake.nix:30]:
```suggestion
```

**Pull request merged**